andela/hc-ravens
# #156687764 #156687763 #156687760 #156687761 Set Up Priority List and Allocation. Create Dashboard For Business Processes

### What does this PR do?

- Sets up a list of users(priority list) to be notified one after the other when a job fails until the alert is resolved.
- Define who in the directory of the team access should be notified about a failed job.
- Allows user to allocate different priorities to different health checks as per a pre-defined priority criteria.
- Allows user to allocate different stakeholders as owners of different HealthCheck alerts.
- Allows teams to create a dashboard of all mission critical automated or automate-able business processes, for example,. IT, Marketing and Sales.

### Description of Task to be completed?
#### Have the following features working.
- A team access account super-user should be able to define who in the directory of the team access should be notified about a failed job.
-  When they should be notified and in what order.
- Users should be able to set the priority level of a job as per a predefined priority criteria. 
- User should be able to invite other team members as owners of different Checks via email.
- Team access should not guarantee all existing individual Health-checks jobs access. 
- Different departments within a team should be able to categorise jobs within their domains to a separate view.

### How should this be manually tested?

- After cloning the repo cd into hc-ravens folder activate virtual env.
- Log in and create two or more checks e.g Check A and Check B.
- Set the Timeout, Grace and Nag sliders to 1 minute each.
- Once logged into the dashboard, click account settings, go to team access.
- As the super user, invite a team member and assigned a defined check e.g Check A.
- Define the scope of the individual Health-checks jobs for which they have team access.
- When the checks go down the superuser should get alerts for both Check A and Check B while the invited user should only get a notification for check A. 
- Users should get notifications for both down and up notifications in the predefined scope.
- As the user, mark a check as High-priority, the check should then appear as the first check in the users list of checks.

### What are the relevant pivotal tracker stories?

#156687764
#156687763
#156687760
#156687761
